src/main/java/com/netgrif/application/engine/configuration/AbstractSecurityConfiguration.java

```diff
@@ -34,6 +39,132 @@ public abstract class AbstractSecurityConfiguration extends WebSecurityConfigure
     @Autowired
     private ApplicationContext context;
 
+    protected void setHeaders(HttpSecurity http) throws Exception {
+        setStrictTransportSecurity(http);
+        setContentSecurityPolicy(http);
+        setXFrameOptions(http);
+        setXXSSProtection(http);
+
+    }
+
+    protected void setStrictTransportSecurity(HttpSecurity http) throws Exception {
+        if (existConfigurationHeaders()
+                && getSecurityConfigProperties().getHeaders().getHsts() != null
+                && getSecurityConfigProperties().getHeaders().getHsts().isEnable()
+                && getSecurityConfigProperties().getHeaders().getHsts().getMaxAge() >= 0) {
+
+            if (Objects.nonNull(getSecurityConfigProperties().getHeaders().getHsts().isIncludeSubDomains())
```

Comment:
can this if else-if else-if tree be refactored, similar to the other methods in this class?